### ISSUE TITLE
.babelrc not being copied over

### ISSUE THREAD

**Issue by Icehunter** [electron-userland/electron-builder#507]
Hi!

Great package here; it seems in the update from 3 -> 5 the build process no longer copies over my .babelrc file which is in the main app folder.

Is there a way to include this? I might have missed it in the docs but didn't see it while reading them over.

Thanks!

**Comment by Icehunter** [comment]
Sorry, figured it out.

```
files: ["**\/*", "**\/.*"]
```

**Comment by develar** [comment]
Reopened— such file must be included. Why do you include it in the final app? Production app should not require babel

**Comment by Icehunter** [comment]
I use it for import/export support:
```
{
  "plugins": ["transform-es2015-modules-commonjs"]
}
```

I guess I could take it out; but that's all I'm using babel for.

**Comment by Icehunter** [comment]
In addition I think I can just add this to package.json and it will be fine. According to the docs here:

https://babeljs.io/docs/usage/babelrc/

**Comment by develar** [comment]
Yes, I suggest to not use babel in production and move config into the package.json.

Currently, your pattern is very permissive — please see https://github.com/electron-userland/electron-builder/pull/496#issuecomment-225784942

**Comment by TimNZ** [comment]
@develar Man why are you hard-coding ignoring certain attributes in extraMetadata?

I'm trying to include a babel section instead of having a .babelrc

Please do not be so opinionated in making decisions for us.
Plus time wasted trying to figure out what is going on..

I have dynamic loading of plugins 
fileTransformer:
```
const ignoredPackageMetadataProperties = new Set(["dist", "gitHead", "keywords", "build", "devDependencies", "scripts", "jspm", "ava", "babel", "xo", "nyc", "eslintConfig"]);
```

**Comment by develar** [comment]
> I have dynamic loading of plugins using babel to support JSX

Why not compile it for production?

**Comment by develar** [comment]
@TimNZ Please provide your `dependencies` We will check that no babel plugins in it and only in this case remove babel config.

**Comment by TimNZ** [comment]
Why are you making it so complex @develar and more work for yourself?  
Why can't you just leave babel in there - what do you care?

**Comment by develar** [comment]
@TimNZ Some users worries about security and don't want to see dev stuff.

**Comment by develar** [comment]
@TimNZ Fixed in the upcoming 19.12.0. `dependencies` list is still welcome.

**Comment by TimNZ** [comment]
Thanks.  
I've ended up using a .babelrc file for the time being.

I don't understand your comment about security - who are these users that care there is a babel section in package.json - who cares what's in there?  How did you decide there is a security concern, and what is it?  Who's looking at package.json, typically inside app.asar?

If you must insist on a dependencies check, anything starting with 'babel', though even that is arbitrary.

**Comment by develar** [comment]
@TimNZ #1764 #1212

**Comment by TimNZ** [comment]
@develar You based on your comments on one person saying babel should be stripped?

That is not a rigorous justification for doing anything unless that person happens to be your boss and they will fire you for not following instructions :)

**Comment by develar** [comment]
You should preprocess for production build and babel should be in the dev deps.

**Comment by TimNZ** [comment]
Focus on improving the QA of this excellent very appreciated tool and stop telling people what they should do.

**Comment by develar** [comment]
@TimNZ that's why your comment is taken into account and it is fixed in the 19.3.0 No — I will continue telling  what people should do — it is called convention over configuration. If I am wrong — thanks for feedback. In this case we still recommend to not use babel in the production but if you want to do — it is ok and does not lead to issue.

QA — we do unit and integration tests, but final QA is performed by users. We use tagged as next releases, but often users don't use it and feedback reported after release tagged as latest.
